src/python/gem5/simulate/simulator.py

```diff
@@ -568,11 +588,10 @@ def run(self, max_ticks: int = m5.MaxTick) -> None:
         This function will start or continue the simulator run and handle exit
         events accordingly.
 
-        :param max_ticks: The maximum number of ticks to execute per simulation
-                          run. If this ``max_ticks`` value is met, a ``MAX_TICK``
+        :param max_ticks: The maximum number of ticks to simulate overall.
+                          If this ``max_ticks`` value is met, a ``MAX_TICK``
                           exit event is received, if another simulation exit
-                          event is met the tick count is reset. This is the
-                          **maximum number of ticks per simulation run**.
+                          event is met the tick count is reset.
```

Comment:
Is this changing the behavior? If so, it concerns me. Either way, the description *still* isn't clear.

Reply:
Yes, it's changing the behavior. I am not hellbent on getting this merged, if you don't see any value in this I can close this.

Reply:
The problem with changing behavior is that there are many scripts out there that expect the old behavior. We could add a new parameter which has this behavior and deprecate `max_ticks`, but we can't just break people's current scripts.

Reply:
I see, I can make that change.